xarray/coding/variables.py

```diff
@@ -212,11 +212,24 @@ class CFScaleOffsetCoder(VariableCoder):
         decode_values = encoded_values * scale_factor + add_offset
     """
 
+    @staticmethod
+    def _choose_float_dtype(data, attributes):
+        # We default to en/decoding as float64, but for small integer types
+        # and no offset it's usually safe to use float32, which saves a lot
+        # of memory for eg. TB-scale satellite imagery collections.
+        if data.dtype.itemsize <= 2 and \
+                np.issubdtype(data.dtype, np.integer) and \
+                'add_offset' not in attributes and \
+                2 ** -23 < float(attributes.get('scale_factor', 1)) < 2 ** 8:
```

Comment:
This if statement is fairly complicated and its not really clear what is happening, can you

1) add some more comments
2) use parentheses instead of line continuation breaks
3) consider just passing in the scale factor instead of attributes